gensim/matutils.py

```diff
@@ -699,13 +699,24 @@ def unitvec(vec, norm='l2'):
             return vec
 
     if isinstance(vec, np.ndarray):
-        vec = np.asarray(vec, dtype=float)
+        vec = np.asarray(vec, dtype=vec.dtype)
         if norm == 'l1':
             veclen = np.sum(np.abs(vec))
         if norm == 'l2':
             veclen = blas_nrm2(vec)
         if veclen > 0.0:
-            return blas_scal(1.0 / veclen, vec)
+            '''EDIT: included .astype(vec.dtype)'''
+            if vec.dtype == 'int16':
```

Comment:
For int-like inputs, IMO we don't necessarily have to match the dtype of the returned array to the corresponding float type (e.g. `int16` -> `float16`). We could simply return a `float` array when the input array is int-like.
It'd also be good to have a more robust way of checking if the input dtype is int-like instead of a hardcoded list of types.

Reply:
Thanks, I'll work on that final point.

Reply:
Hi, I edited the branch that I used for this to make changed to unitvec and present tests, but how do I get it to show up?

Reply:
@o-P-o you should to push the code to `patch-1` branch in your fork.